Flaky test failure

### Description

```
Run mix test
Compiling 21 files (.ex)
Generated sentry app
Wrote 90 files in 498.09 kb to: _build/test/lib/sentry/priv/sentry.map
Running ExUnit with seed: 476905, max_cases: 8

.....................................................................................................................................................................................................................................................................................................................................................................................................................

  1) test send_event/2 uses the async sender pool when :result is :none (Sentry.ClientTest)
Error:      test/sentry/client_test.exs:378
     Assertion failed, no matching message after 5000ms
     The following variables were pinned:
       ref = #Reference<0.4108024795.1505230851.237813>
     The process mailbox is empty.
     code: assert_receive {^ref, body}
     stacktrace:
       test/sentry/client_test.exs:393: anonymous fn/1 in Sentry.ClientTest."test send_event/2 uses the async sender pool when :result is :none"/1
       (elixir 1.18.4) lib/stream.ex:1505: Stream.do_repeatedly/3
       (elixir 1.18.4) lib/stream.ex:1773: Enumerable.Stream.do_each/4
       (elixir 1.18.4) lib/enum.ex:4621: Enum.slice_enum/5
       (elixir 1.18.4) lib/enum.ex:479: Enum.at/3
       test/sentry/client_test.exs:400: (test)

...............................................................................................................................
Finished in 18.9 seconds (2.8s async, 16.0s sync)
15 doctests, 518 tests, 1 failure
Error: Process completed with exit code 2.
```

--- a/config/config.exs
+++ b/config/config.exs
@@ -13,6 +13,8 @@ if config_env() == :test do
     test_mode: true,
     traces_sample_rate: 1.0
 
+  config :sentry, request_retries: []
+
   if System.version() > "1.16.0" do
     config :logger, :default_handler, false
 

--- a/lib/sentry/transport/sender.ex
+++ b/lib/sentry/transport/sender.ex
@@ -60,10 +60,12 @@ defmodule Sentry.Transport.Sender do
 
   @impl GenServer
   def handle_cast({:send, client, %Event{} = event, counter_key}, %__MODULE__{} = state) do
+    retries = Application.get_env(:sentry, :request_retries, Transport.default_retries())
+
     _ =
       event
       |> Envelope.from_event()
-      |> Transport.encode_and_post_envelope(client)
+      |> Transport.encode_and_post_envelope(client, retries)
 
     # We sent an event, so we can decrease the number of queued events.
     Transport.SenderPool.decrease_queued_events_counter(counter_key)
@@ -76,10 +78,12 @@ defmodule Sentry.Transport.Sender do
         {:send, client, %Transaction{} = transaction, counter_key},
         %__MODULE__{} = state
       ) do
+    retries = Application.get_env(:sentry, :request_retries, Transport.default_retries())
+
     _ =
       transaction
       |> Envelope.from_transaction()
-      |> Transport.encode_and_post_envelope(client)
+      |> Transport.encode_and_post_envelope(client, retries)
 
     # We sent a transaction, so we can decrease the number of queued transactions.
     Transport.SenderPool.decrease_queued_transactions_counter(counter_key)

--- a/test/mix/sentry.send_test_event_test.exs
+++ b/test/mix/sentry.send_test_event_test.exs
@@ -66,13 +66,6 @@ defmodule Mix.Tasks.Sentry.SendTestEventTest do
       Plug.Conn.resp(conn, 500, ~s<{"id": "340"}>)
     end)
 
-    original_retries =
-      Application.get_env(:sentry, :request_retries, Sentry.Transport.default_retries())
-
-    on_exit(fn -> Application.put_env(:sentry, :request_retries, original_retries) end)
-
-    Application.put_env(:sentry, :request_retries, [])
-
     put_test_config(dsn: "http://public:secret@localhost:#{bypass.port}/1")
 
     assert_raise Mix.Error, ~r/Error sending event/, fn ->

--- a/test/sentry/client_report/sender_test.exs
+++ b/test/sentry/client_report/sender_test.exs
@@ -7,13 +7,6 @@ defmodule Sentry.ClientReportTest do
   alias Sentry.Event
 
   setup do
-    original_retries =
-      Application.get_env(:sentry, :request_retries, Sentry.Transport.default_retries())
-
-    on_exit(fn -> Application.put_env(:sentry, :request_retries, original_retries) end)
-
-    Application.put_env(:sentry, :request_retries, [])
-
     bypass = Bypass.open()
     put_test_config(dsn: "http://public:secret@localhost:#{bypass.port}/1")
     %{bypass: bypass}

